File: solutions/typescript/004-median-of-two-sorted-arrays.ts
/**
 * Problem 4: Median of Two Sorted Arrays (LeetCode 4)
 * Difficulty: Hard
 * Language: TypeScript
 */
function findMedianSortedArrays(nums1: number[], nums2: number[]): number {
  const merged = [...nums1, ...nums2].sort((a, b) => a - b);
  const mid = Math.floor(merged.length / 2);
  return merged.length % 2 === 0
    ? (merged[mid - 1] + merged[mid]) / 2
    : merged[mid];
}

File: solutions/clojure/004-median-of-two-sorted-arrays.clj
;; Problem 4: Median of Two Sorted Arrays (LeetCode 4)
;; Difficulty: Hard
;; Language: Clojure
;; 
(defn find-median-sorted-arrays [nums1 nums2]
  (let [merged (sort (concat nums1 nums2))
        n (count merged)
        mid (quot n 2)]
    (if (even? n)
      (/ (+ (nth merged (dec mid)) (nth merged mid)) 2.0)
      (double (nth merged mid)))))
